### PR TITLE
use k8s-staging-test-infra/gcb-docker-gcloud

### PR DESCRIPTION
**Pull request by spiffxp**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Related:
- Part of: https://github.com/kubernetes/k8s.io/issues/1523
- Followup to: https://github.com/kubernetes/test-infra/pull/23656

Use a version of gcb-docker-cloud in cloudbuild.yaml that is hosted in a
community-owned repo instead of a google.com-owned repo

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/krel/cmd/testdata/changelog-v1.18.0-alpha.3/GetPullRequest-158.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
...
...s required:\r\nEnter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string \"action required\".\r\n\r\nFor more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md\r\n--\u003e\r\n```release-note\r\nNONE\r\n```\r\n\r\n**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:\r\n\r\n\u003c!--\r\nThis section can be blank if this pull request does not require a release note.\r\n\r\nWhen adding links which point to resources within git repositories, like\r\nKEPs or supporting documentation, please reference a specific commit and avoid\r\nlinking directly to the master branch. This ensures that links reference a\r\nspecific point in time, rather than a document that may change over time.\r\n\r\nSee here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files\r\n\r\nPlease use the following format for linking documentation:\r\n- [KEP]: \u003clink\u003e\r\n- [Usage]: \u003clink\u003e\r\n- [Other doc]: \u003clink\u003e\r\n--\u003e\r\n```docs\r\n\r\n```\r\n",
```
Comment (puerco) [inline cmd/krel/cmd/testdata/changelog-v1.18.0-alpha.3/GetPullRequest-158.json:8]: Funny, why did this line change? Besides the newline at the end, I don't see a difference. If the test keeps passing it is not important of course, but its strange isn't it?

Comment (spiffxp) [reply cmd/krel/cmd/testdata/changelog-v1.18.0-alpha.3/GetPullRequest-158.json:8]: There was a reference to `k8s-testimages` that my search-replace caught... if this ends up validating against live data I'll revert the change

Comment (puerco) [reply cmd/krel/cmd/testdata/changelog-v1.18.0-alpha.3/GetPullRequest-158.json:8]: It should not be a problem, this is just replay data for the test so it should be ok :)

Comment (puerco) [reply cmd/krel/cmd/testdata/changelog-v1.18.0-alpha.3/GetPullRequest-158.json:8]: Ah I see it now:
image

**Review by puerco — APPROVED** [review]
Thanks @spiffxp  !
/lgtm
/approve

**Comment by spiffxp** [comment]
Looks like this is https://github.com/kubernetes/test-infra/issues/20652
```
"error": {
    "code": 429,
    "message": "Quota exceeded for quota metric 'Build and Operation Get requests' and limit 'Build and Operation Get requests per minute' of service 'cloudbuild.googleapis.com' for consumer 'project_number:86929635859'.",
    "status": "RESOURCE_EXHAUSTED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "RATE_LIMIT_EXCEEDED",
        "domain": "googleapis.com",
        "metadata": {
          "consumer": "projects/86929635859",
          "quota_metric": "cloudbuild.googleapis.com/get_requests",
          "service": "cloudbuild.googleapis.com",
          "quota_limit": "GetRequestsPerMinutePerProject"
        }
      }
    ]
  }
```

**Pull request merged**